src/CoreArbiterServer.cc

```diff
@@ -1558,6 +1565,12 @@ void signalHandler(int signum) {
         }).detach();
     } else if ((signum == SIGSEGV) || (signum == SIGABRT)) {
         invokeGDB(signum);
+    } else if (signum == SIGPIPE) {
+        // Ignore SIGPIPE and allow repeated invocations
+        signalAction.sa_handler = signalHandler;
+        sigaction(signum, &signalAction, NULL);
+        LOG(WARNING, "Received SIGPIPE: connection was not closed normally");
```

Comment:
Code in a signal handler is very dangerous: it's effectively unsynchronized with respect to the rest of the application (the signal can happen at any time, no?). Let's discuss whether this particular code is safe.  Also, I think there may be a way that you can stop SIGPIPE signals and just get errors on I/O operations instead. If so, I'd recommend using this approach.